File: Java/jax-rs-models/src/main/java/com/untitleddoc/cadencecalc/jaxrs/models/Crankset.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.untitleddoc.cadencecalc.jaxrs.models;

import com.google.common.collect.Lists;
import java.io.Serializable;
import java.util.List;
import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.AllArgsConstructor;
import lombok.Data;

/**
 *
 * @author Novikov
 */
@XmlRootElement
@Data
@AllArgsConstructor
public class Crankset implements Serializable, IC2Model {

    private static final long serialVersionUID = -6512261512456314356L;

    private String name;
    private String manufacture;
    private List<Integer> tooths;

    public Crankset() {
	}

	@Override
    public String displayValue() {
        final List<String> stooths = Lists.transform(tooths, String::valueOf);
        return getName() + " (" + String.join("-", stooths) + ")";
    }
}

File: jax-rs-models/src/main/kotlin/Crankset.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/*

  @author k-aki
  Created on 2018/06/03
*/

package com.untitleddoc.cadencecalc.jaxrs.models.kotlin

import com.untitleddoc.cadencecalc.jaxrs.models.IC2Model
import java.io.Serializable
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement
public data class Crankset(val name: String, val manufacture: String, val tooths: List<Int>): Serializable, IC2Model {
    public override fun displayValue(): String = name + tooths.joinToString("-", " (", ")")
}